<a href="https://colab.research.google.com/github/gilesitorr/Estadistica_Avanzada/blob/main/Parcial_2/Tarea_2/EA_parcial2_tarea2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# __Estadística Avanzada__
# Parcial 2 - Tarea 2
## _Gil Estéfano Rodríguez Rivera_

Todas las bases de datos utilizadas así como el desarrollo se encuentra en _GitHub_ en https://github.com/gilesitorr/Estadistica_Avanzada/tree/main/Parcial_2/Tarea_2

__Nota:__ En este Notebook se usó el kernel de R, aunque la extensión sea ipynb.

## Biostatistics: A Foundation for Analysis in the Health Sciences
### Wayne W. Daniel & Chad L. Cross

### 10.3.1


Obtain the regression equation for each of the following data set

__Machiel Naeije (A-2) studied the relationship between maximum mouth opening and measurements of the lower jaw (mandible). He measured the dependent variable, maximum mouth opening (MMO, measured in mm), as well as predictor variables, mandibular length (ML, measured in mm) and angle of rotation of the mandible (RA, measured in degrees) of 35 subjects.__

In [1]:
# Se define el url donde está guardado el archivo csv a usar
url = "https://github.com/gilesitorr/Estadistica_Avanzada/raw/main/Parcial_2/Tarea_2/EXR_C10_S03_01.csv"

# Se lee el archivo csv desde el url que se definió
df <- read.csv(url, header=TRUE)

In [2]:
# Se realiza la regresión múltiple
ajuste <- lm(Y~X1+X2, data=df)
ajuste


Call:
lm(formula = Y ~ X1 + X2, data = df)

Coefficients:
(Intercept)           X1           X2  
   -31.4248       0.4732       1.0712  


La ecuación de la regresión múltiple es:

_$Y = 0.4732 X_1 + 1.0712 X_2 - 31.4248$_

### 10.4.1


__Refer to Exercise 10.3.1.__

(a) Calculate the coefficient of multiple determination;

(b) perform an analysis of variance;

(c) test the significance of each beta_i (i > 0) . Let a = 0.05 for all tests of
significance and determine the p value for all tests;

(d) construct a 95 percent confidence interval for each significant sample slope.

Se utilizará el trabajo realizado en el ejercicio anterior (no es necesario utilizar otra variable para el dataframe ni para el ajuste de la regresión así como no es necesario cargar más información)

In [3]:
summary(ajuste)


Call:
lm(formula = Y ~ X1 + X2, data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.6219 -0.9816  0.1288  1.2022  6.2728 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) -31.42480    6.14747  -5.112 1.44e-05 ***
X1            0.47317    0.06117   7.736 8.05e-09 ***
X2            1.07117    0.06280  17.058  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 2.064 on 32 degrees of freedom
Multiple R-squared:  0.9204,	Adjusted R-squared:  0.9155 
F-statistic: 185.1 on 2 and 32 DF,  p-value: < 2.2e-16


El __coeficiente de determinación múltiple__ es _$R^2=0.9204$_.

De acuerdo con la prueba F, puede verse que, con una significancia del 5%, __al menos una de las pendientes de la recta, beta_1 y beta_2, son distintas de cero__, con p-value _$p < 2 \times 10^{-16}$_.

De acuerdo a las prueba t bilateral, con una significancia del 5%, tanto __$\beta_1$ como $\beta_2$ son diferentes de 0__, con p-values, respectivamente, $p_1=1.44\times10^{-5}$ y $p_2=8.05\times10^{-9}$.

In [4]:
# Se calcula el intervalo de confianza del 95% de beta_1
confint(ajuste, level=0.95)

2.5 %       97.5 %     
(Intercept) -43.9467785 -18.9028184
X1            0.3485846   0.5977639
X2            0.9432598   1.1990851

El __intervalo de confianza__ del 95% de cada pendiente son:

$\beta_1$ : (0.3486, 0.5978)

$\beta_2$ : (0.9433, 1.1991)

### 10.5.1


__Refer to Exercise 10.3.1 and let $x_{1j}$=95 and $x_{2j}$=35:.__

For each of the following exercises compute the y value and construct

(a) 95 percent confidence

(b) 95 percent prediction intervals for the specified values of Xi.

Se utilizará el trabajo realizado en el ejercicio 10.3.1 (no es necesario utilizar otra variable para el dataframe ni para el ajuste de la regresión así como no es necesario cargar más información)

In [5]:
# Se construyen los intervalos de confianza del 95% para X=1.6
predict(ajuste, data.frame(X1=95, X2=35), interval="confidence")

fit      lwr      upr     
1 51.01779 50.28862 51.74696

In [6]:
# Se construyen los intervalos de predicción del 95% para X=1.6
predict(ajuste, data.frame(X1=95, X2=35), interval="prediction")

fit      lwr      upr     
1 51.01779 46.75122 55.28436

La predicción para $X_1=95$, $X_2=35$ es $y=51.0178$.

Este valor predicho tiene dos intervalos asociados (ambos del 95%).

1) El de __confianza__ es de (50.2886, 51.7470)

2) El de __predicción__ es de (46.7512, 55.2844).

### 10.6.1

__The objective of a study by Anton et al. (A-5) was to investigate the correlation structure of multiple measures of HIV burden in blood and tissue samples. They measured HIV burden four ways. Two measurements were derived from blood samples, and two measurements were made on rectal tissue. The two blood measures were based on HIV DNA assays and a second co-culture assay that was a modification of the first measure. The third and fourth measurements were quantitations of HIV-1 DNA and RNA from rectal biopsy tissue. The table below gives data on HIV levels from these measurements for 34 subjects.__

(a) Compute the simple correlation coefficients between all possible pairs of variables.

(b) Compute the multiple correlation coefficient among the four variables. Test the overall correlation for significance.

(c) Calculate the partial correlations between HIV DNA blood and each one of the other variables while controlling for the other two. (These are called second-order partial correlation coefficients.)

(d) Calculate the partial correlation between HIV co-culture blood and HIV DNA, controlling for the other two variables.

(e) Calculate the partial correlation between HIV co-culture blood and HIV RNA, controlling for the other two variables.

(f) Calculate the partial correlations between HIV DNA and HIV RNA, controlling for the other two variables.

In [7]:
# Se define el url donde está guardado el archivo csv a usar
url = "https://github.com/gilesitorr/Estadistica_Avanzada/raw/main/Parcial_2/Tarea_2/EXR_C10_S06_01.csv"

# Se lee el archivo csv desde el url que se definió
df <- read.csv(url, header=TRUE)

In [8]:
# Se aculcula el coeficiente de correlación de Pearson entre ambas variables
# de la muestra
(r <- cor(df$Y, df$X1))
(r <- cor(df$Y, df$X2))
(r <- cor(df$Y, df$X3))
(r <- cor(df$X1, df$X2))
(r <- cor(df$X1, df$X3))
(r <- cor(df$X2, df$X3))

[1] 0.3604168

[1] 0.5320713

[1] 0.2022809

[1] 0.3025645

[1] 0.6740663

[1] 0.4295396

Los __coeficientes de correlación simples__ de cada par de variables son:

$Y$, $X_1$ : 0.3604

$Y$, $X_2$ : 0.5321

$Y$, $X_3$ : 0.2023

$X_1$, $X_2$ : 0.3026

$X_1$, $X_3$ : 0.6741

$X_2$, $X_3$ : 0.4295

In [20]:
# Para calcular el coeficiente de correlación múltiple, se hace el ajuste
# lineal, se calcula el summary del ajuste y se extrae el valor del R^2
ajuste <- lm(Y~X1+X2+X3, data=df)
summary(ajuste)


Call:
lm(formula = Y ~ X1 + X2 + X3, data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-76.923 -26.748  -8.207  18.544 246.083 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 27.79546   12.90719   2.153 0.038053 *  
X1          62.58296   28.17667   2.221 0.032724 *  
X2           0.04801    0.01303   3.683 0.000751 ***
X3          -0.06229    0.03956  -1.574 0.124133    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 54.08 on 36 degrees of freedom
Multiple R-squared:  0.3702,	Adjusted R-squared:  0.3178 
F-statistic: 7.055 on 3 and 36 DF,  p-value: 0.0007527


$R^2=0.3702$

El __coeficiente de correlación múltiple__ es $R=0.6085$. (Se toma la solución positiva porque la correlación individual de Y con las $X_i$ es positiva). De acuerdo a una prueba __F__, se infiere que con una significancia menor a 5%, se tieneal menos una de las pendientes $\beta_i$, i=1,2,3 es diferente de cero, con un p-value de $p=0.0008$.

In [13]:
# La libreria para estudiar correlaciones parciales
install.packages("ppcor")
library(ppcor)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Loading required package: MASS



In [21]:
# La matriz de correlaciones parciales
pcor(df)$estimate

Y          X1         X2         X3        
Y   1.0000000  0.3471589  0.5231596 -0.2538148
X1  0.3471589  1.0000000 -0.1660120  0.6615325
X2  0.5231596 -0.1660120  1.0000000  0.3969161
X3 -0.2538148  0.6615325  0.3969161  1.0000000

De la matriz de __correlaciones parciales__ puede verse que la correlación que guarda $Y$ con cada una de las variables es la siguiente:

$Y$, $X_1$ : 0.3472

$Y$, $X_2$ : 0.5232

$Y$, $X_3$ : -0.2538

La __correlación parcial__ entre $X_1$ y $X_2$ es $-0.1660$.

La __correlación parcial__ entre $X_1$ y $X_3$ es $0.6615$.

La __correlación parcial__ entre $X_2$ y $X_3$ es $0.3969$.